### DISCUSSION TITLE
Publish to crates.io?

### DISCUSSION THREAD

**Discussion by jmjoy** [lapce/lapce#516] [Ideas]
So we can install it by `cargo install`.

**Comment by ket000** [comment]
Yes, this is a good idea.

**Comment by kevinmatthes** [comment]
I agree that a release on the official package registry is a good idea.  I would like to add that an installation using `cargo` is always possible as long there is a public source to fetch from.  The command below will directly install the software into the user's home directory after building it freshly from source.  Hence, until the release to the registry is done, this command might be a sane workaround.

```
cargo install --git https://github.com/lapce/lapce
```

For further information, please consult the official documentation of `cargo` for the `install` sub-command.

**Comment by ket000** [comment]
Yes, I agree with the workaround. However installation into public registry will make sure that there are no security issues and may be more easily discoverable by other users.

**Comment by panekj** [comment]
Lapce will not be published to crates.io, crates.io is not designed to distribute software such as Lapce and we don't want to support installation from crates.io.

**Reply by jmjoy** [reply]
I think that since it can be installed from the source code by `cargo build`, so `cargo install` from `crates.io` should be unimpeded.

**Comment by ket000** [comment]
This is really sad as this makes users to use alternate methods like
fetching it from github. I am exactly not sure what is the issue with
publishing lapce on crates.io lile all other rust based software.

**Reply by panekj** [reply]
It doesn't give/provide any benefit, and lapce isn't just rust-based software, it's GUI application that depends on OS platform SDK and not some CLI tool that happens to build successfully.
crates.io is not a binary software distribution registry.

> This is really sad as this makes users to use alternate methods like fetching it from github.

You are supposed to use your system package manager to install Lapce.

**Comment by ket000** [comment]
System package manager requires root privileges. I will use --git option as
it works without root privilege as a workaround.

**Comment by jkemp814** [comment]
I'm getting this error message:

cargo install --git https://github.com/lapce/lapce

error: multiple packages with binaries found: lapce, lapce-proxy, lapce-ui. When installing a git repository, cargo will always search the entire repo for any Cargo.toml. Please specify which to install.

Shouldn't it install all three?

Any guidance would be greatly appreciated.
